### PR TITLE
Initial code for sign out

### PR DESCRIPTION
**Pull request by bzvestey**
https://onelogin2.atlassian.net/browse/LAUN-518

I have not tested this yet, but getting it up for early review and so that it is available.

### CODE REVIEW

File: oneloginoidc/src/main/java/com/onelogin/oidc/data/network/NetworkClient.kt [hunk 1]
```diff
@@ -16,6 +16,26 @@ internal class NetworkClient(
     private val configuration: OIDCConfiguration
 ) {
 
+    suspend fun logout(token: String) = withContext(Dispatchers.IO) {
+        val request = Request.Builder()
+            .url(configuration.issuer + EndpointsContract.LOGOUT_PATH)
+            .header("Content-Type", "application/x-www-form-urlencoded")
+            .header("Authorization", "Bearer $token")
+            .build()
```
Comment (dyrczyk) [inline oneloginoidc/src/main/java/com/onelogin/oidc/data/network/NetworkClient.kt:24]: Please check the usage of /oidc/2/logout
https://developers.onelogin.com/openid-connect/api/logout
I do not think that `"Authorization", "Bearer $token"` is the correct value here.
It should be "id_token_hint" instead if id_token is stored somewhere before.

Comment (bzvestey) [reply oneloginoidc/src/main/java/com/onelogin/oidc/data/network/NetworkClient.kt:24]: Ah, ok. I was a little confused with the documentation, and I haven't had a change to test this code yet. I will look into that.

File: oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManager.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.onelogin.oidc.logout
+
+import android.app.Activity
+import com.onelogin.oidc.Callback
+
+interface SignOutManager {
+    suspend fun singOut(
```
Comment (kdallan-ol) [inline oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManager.kt:7]: I like the idea of signing out. Should this be signOut though?

File: oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManagerImpl.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import net.openid.appauth.AuthorizationServiceConfiguration
+import net.openid.appauth.EndSessionRequest
+
+internal class SignOutManagerImpl(
+    private val configuration: OIDCConfiguration,
+    private val repository: OIDCRepository,
+    private val signOutFragmentProvider: (EndSessionRequest) -> SignOutFragment
+) : SignOutManager {
+
+    override suspend fun singOut(
```
Comment (kdallan-ol) [inline oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManagerImpl.kt:19]: signOut?

**Review by kdallan-ol — APPROVED** [review]

File: oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManager.kt (hunk 2 above)
Comment (bzvestey) [reply oneloginoidc/src/main/java/com/onelogin/oidc/logout/SignOutManager.kt]: I mean, its a word and I am consistent with the spelling 😄

**Review by jcbarnett — APPROVED** [review]

**Pull request merged**